lib/Parse/Lexer.cpp

```diff
@@ -1558,14 +1657,34 @@ void Lexer::tryLexEditorPlaceholder() {
 }
 
 StringRef Lexer::getEncodedStringSegment(StringRef Bytes,
-                                         SmallVectorImpl<char> &TempString) {
+                                         SmallVectorImpl<char> &TempString,
+                                         bool MultilineString,
+                                         bool IsFirstSegment,
+                                         bool IsLastSegment,
+                                         unsigned IndentToStrip) {
+
   TempString.clear();
   // Note that it is always safe to read one over the end of "Bytes" because
   // we know that there is a terminating " character.  Use BytesPtr to avoid a
   // range check subscripting on the StringRef.
   const char *BytesPtr = Bytes.begin();
-  while (BytesPtr != Bytes.end()) {
+  while (BytesPtr < Bytes.end()) {
```

Comment:
Shoun’t be by < is safer particularly since we are adding IndentToStrip to BytesPtr